### ISSUE TITLE
[New build - EXPERIMENTAL] 21w13a: Spaceship and build SFX

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#3698]
This builds adds some revamped Spaceship, hovercraft and building sounds effects.

Improvements for spaceship sfx:
- more connected to player input
- Deals better with perspective
- Account for ship size.

Bugfixes:
#3689 Symmetry overlap check do not work sometime.
#3697 Ship fall through the ground when far away.

**Comment by tsunamayo** [comment]
Build is up! Thanks all

**Comment by Kaiser-Indrasil** [comment]
The engine sound reminds me of the [Cyclops submarine from Subnautica](https://youtu.be/wPnhbhxMgsQ?t=42)!

**Comment by Regnion** [comment]
I don't like the engine sounds at all. It sounds like what I imagine a little hover scooter would sound like. For fighters, for large ships, the sound just doesn't have _oomph_. While I understand you don't exactly have a sound designer and an entire workshop available to you for mixing sound effects, I'd like to provide some examples.

<details>
<summary> MILLENNIUM FALCON SOUND </summary>
https://www.youtube.com/watch?v=snNxffxAMaA

<summary> STAR DESTROYER SOUND </summary>
https://www.youtube.com/watch?v=JxDH4GPPmcg

<summary> PROPELLER AIRCRAFT SOUND </summary>
https://www.youtube.com/watch?v=zl4KdIa_gsw

<summary> SATURN V SOUND </summary>
https://youtu.be/ViNcBQ8cDA0?t=42
</details>

**Comment by ZachZent** [comment]
I partially agree with Reg. The engine sound should be a bit more beefy, but still keep the sci-fi engine sound. Maybe engine deepness can scale with size. Small engines making a higher pitched sound (relatively) while larger stretched block engines make a deeper/low pitched sound.

**Comment by SovietSponge** [comment]
I gotta be honest, I really dislike the new sounds, ships sound like roombas now. It really just doesn't at all fit what you're seeing with most ships, on larger ships where you expect there to be a big rumble there just isn't one and on fighters and such well they damn near sound like mosquitoes.

**Comment by TIKIRobo** [comment]
I think also the type of thruster could make a different noise, the small thrusters seem to be electric so they should make a different noise from the fuel burning LC thrusters

**Comment by tsunamayo** [comment]
Okay I have just realized that the thruster sounds where not playing correctly, I fixed it for next patch. Thruster sounds just like S-5 thruster, depending on whether you are in atmosphere or not the burn sound will be more or less pronounced.
Regarding the scale impact it just control how far the sounds goes for now, I will see what I could do to impact with scale.
Things is a larger ship is not necessarily louder... I cant use a different set of sounds too, it have to be a continuous thing.
Also I cant do a thing that shake your skull. It sounds good for a trailer but it would be just plain painful for long game session. It have to be a smooth thing.
The sounds I was not liking was more the destroy, what do you think of it?

**Comment by Kaiser-Indrasil** [comment]
I've played dozens of sci-fi games, many of which were space fighter sims and none ever came close to how good [the engines in Elite: Dangerous](https://www.youtube.com/watch?v=UKGR2FLUqJw) sound like. Enjoy.

**Comment by Kaiser-Indrasil** [comment]
Btw did you think of using something like a sound gradient? (not a technical term) It would gradually shift from one pitch to the next, scaling with the amount of thrust a given ship has. It could pull that data directly from the Spaceship Core block.

You could have, say, 5 different key pitches that would blend together. For example, if the ship had combined thrust which would place it between the pitch 3 and 4 (favouring the 4th), it would have an blend of engine sounds at 40% pitch 3 and 60% pitch 4.

**Comment by tsunamayo** [comment]
yeah there is plenty of pitch and volume scaling already. I will tweak it a bit more.

**Comment by tsunamayo** [comment]
what is cool in the Elite one is that it depend on the engine intertia, ie if you throttle up, it will sound loud when the engine is catching up, but once the speed reached is become quieter.

**Comment by Regnion** [comment]
> 
> 
> Okay I have just realized that the thruster sounds where not playing correctly, I fixed it for next patch. Thruster sounds just like S-5 thruster, depending on whether you are in atmosphere or not the burn sound will be more or less pronounced.
> Regarding the scale impact it just control how far the sounds goes for now, I will see what I could do to impact with scale.
> Things is a larger ship is not necessarily louder... I cant use a different set of sounds too, it have to be a continuous thing.
> Also I cant do a thing that shake your skull. It sounds good for a trailer but it would be just plain painful for long game session. It have to be a smooth thing.
> The sounds I was not liking was more the destroy, what do you think of it?

Is there any way to have directional sound? Or, positional? Perhaps inside the ship, the engine sound is a low hum, like the propeller plane, but behind the ship, it has a much more violent sound. This would also make fly-by's really impressive. This might also be rather complicated, I'm not a game dev.

**Comment by tsunamayo** [comment]
@Regnion yeah they are supposed to be like that actually, but there was an issue in the build. Will be fixed for next build which drop soon. It has some glow block so hopefully you will like it.
